borg/testsuite/helpers.py

```diff
@@ -104,6 +104,36 @@ def test(self):
         self.assert_equal(self.evaluate(['/etc/', '/var'], ['dmesg']),
                           ['/etc/passwd', '/etc/hosts', '/var/log/messages', '/var/log/dmesg'])
 
+class OSXPatternNormalizationTestCase(BaseTestCase):
+    def testComposedUnicode(self):
+        pattern = 'b\N{LATIN SMALL LETTER A WITH ACUTE}'
+        i = IncludePattern(pattern)
+        ni = IncludePattern(pattern, normalize=True)
+
+        assert i.match("b\N{LATIN SMALL LETTER A WITH ACUTE}/foo")
+        assert not i.match("ba\N{COMBINING ACUTE ACCENT}/foo")
+        assert ni.match("b\N{LATIN SMALL LETTER A WITH ACUTE}/foo")
+        assert ni.match("ba\N{COMBINING ACUTE ACCENT}/foo")
+    
+    def testDecomposedUnicode(self):
+        pattern = 'ba\N{COMBINING ACUTE ACCENT}'
+        i = IncludePattern(pattern)
+        ni = IncludePattern(pattern, normalize=True)
+
+        assert not i.match("b\N{LATIN SMALL LETTER A WITH ACUTE}/foo")
+        assert i.match("ba\N{COMBINING ACUTE ACCENT}/foo")
+        assert ni.match("b\N{LATIN SMALL LETTER A WITH ACUTE}/foo")
+        assert ni.match("ba\N{COMBINING ACUTE ACCENT}/foo")
+    
+    def testInvalidUnicode(self):
+        pattern = str(b'ba\x80', 'latin1')
+        i = IncludePattern(pattern)
+        ni = IncludePattern(pattern, normalize=True)
+
+        assert not i.match("ba/foo")
+        assert i.match(str(b"ba\x80/foo", 'latin1'))
+        assert not ni.match("ba/foo")
+        assert ni.match(str(b"ba\x80/foo", 'latin1'))
```

Comment:
above, you decode the bytestrings "correctly" with latin1 (as all codes are mapped in latin1, every byte sequence is correct). did you check how borg handles this? does it work with bytes? does it try to decode somehow?